src/table/tableBody.jsx

```diff
@@ -1,19 +1,29 @@
-import React, { useReducer, useEffect } from 'react';
+import React, { useReducer, useState, useEffect } from 'react';
 import PropTypes from 'prop-types';
 import TableBody from '@material-ui/core/TableBody';
 import TableRow from '@material-ui/core/TableRow';
 import TableCell from '@material-ui/core/TableCell';
 import { addSelectionListeners, reducer } from './selections-utils';
+import getCellRenderer from './cells/renderer';
 
-export default function TableBodyWrapper({ tableData, columnRenderers, selectionsAPI }) {
+export default function TableBodyWrapper({ tableData, constraints, selectionsAPI }) {
   const { rows, columns } = tableData;
 
+  const [columnRenderers, setColumnRenderers] = useState([]);
   const [selState, selDispatch] = useReducer(reducer, {
     api: selectionsAPI,
     rows: [],
     colIdx: -1,
+    isEnabled: !!selectionsAPI && !constraints.active,
   });
 
+  useEffect(() => {
+    console.log('setting columns and isEnabled');
+    const isEnabled = !!selectionsAPI && !constraints.active;
+    selDispatch({ type: 'set-enabled', payload: { isEnabled } });
+    setColumnRenderers(tableData.columns.map((c) => getCellRenderer(c, isEnabled)));
+  }, [selectionsAPI, constraints]);
```

Comment:
I can't think of a case where the selection api is suddenly becomes unavailable, without the layout changing and the entire extension rerenders. If that is the case, `selectionsAPI` is not needed in the dependency array